contracts/script/FeeContractWithDefender.s.sol

```diff
@@ -0,0 +1,61 @@
+pragma solidity ^0.8.20;
+
+import { Script } from "forge-std/Script.sol";
+import {
+    ProposeUpgradeResponse,
+    Defender,
+    ApprovalProcessResponse
+} from "openzeppelin-foundry-upgrades/Defender.sol";
+import { Upgrades, Options } from "openzeppelin-foundry-upgrades/Upgrades.sol";
+import { FeeContract as FC } from "../src/FeeContract.sol";
+
+contract FeeContractDefenderDeployScript is Script {
+    function run() public returns (address payable proxy, address multisig) {
+        bytes32 contractSalt = bytes32(abi.encodePacked(vm.envString("FEE_CONTRACT_SALT")));
+
+        ApprovalProcessResponse memory upgradeApprovalProcess = Defender.getUpgradeApprovalProcess();
+        multisig = upgradeApprovalProcess.via;
+
+        vm.startBroadcast(multisig);
+
+        if (upgradeApprovalProcess.via == address(0)) {
+            revert(
+                string.concat(
+                    "Upgrade approval process with id ",
+                    upgradeApprovalProcess.approvalProcessId,
+                    " has no assigned address"
+                )
+            );
+        }
+
+        Options memory opts;
+        opts.defender.useDefenderDeploy = true;
+        opts.defender.salt = contractSalt;
+
+        proxy = payable(
+            Upgrades.deployUUPSProxy(
+                "FeeContract.sol", abi.encodeCall(FC.initialize, (multisig)), opts
+            )
+        );
+
+        vm.stopBroadcast();
+
+        return (proxy, multisig);
+    }
+}
+
+contract FeeContractDefenderUpgradeScript is Script {
+    function run() public returns (string memory proposalId, string memory proposalUrl) {
+        bytes32 contractSalt = bytes32(abi.encodePacked(vm.envString("FEE_CONTRACT_SALT")));
+        Options memory opts;
+        opts.defender.useDefenderDeploy = true;
+        opts.defender.salt = contractSalt;
+        address proxyAddress = 0x61B4C96475B99A6ce01AfF0da7910605D048c125;
```

Comment:
I manually added it for this step and this will change for our actual deployment - when using openzeppelin defender I can't put a parameter in the run() function.

Reply:
the proxy Address is now retrieved from the output logs. see https://github.com/EspressoSystems/espresso-sequencer/commit/a4d119f2de16927fc7fc64fcd2d381d561983f16 and 
https://github.com/EspressoSystems/espresso-sequencer/commit/e458801a694c20792712fc799516aa91d5669557